### PR TITLE
[Ppd 221] iuv generator

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- created a new function for the generation of the iuv 

#### Motivation and Context

- assigned task

#### How Has This Been Tested?

Manual Test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmAuxDigit.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * Generate the IUV Code.<br/>
+     * IUV (17 digits) = &lt;codice segregazione (2n)&gt;&lt;IUV base (max
+     * 13n)&gt;&lt;IUV check digit (2n)&gt;
+     * @throws IuvGeneratorException 
+     */
+    @Override
+    public String generate(Long segregationCode) throws IuvGeneratorException {
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmAuxDigit.java]: IuvGeneratorException is never thrown

File: src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvCodeBusiness.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Generates a iuv <code>iuv</code>
+     * 
+     * @param segregationCode
+     * @return the <code>iuv</code>
+     * @throws IuvGeneratorException 
+     * 
+     */
+    public static String generateIUV(Long segregationCode, Long auxDigit) throws IuvGeneratorException {
+        IuvAlghoritmGenerator iuvGenerator = new IuvAlghoritmGenerator.Builder().build(auxDigit);
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvCodeBusiness.java]: this is not the standard [builder pattern](https://dzone.com/articles/design-patterns-the-builder-pattern#:~:text=We%20can%20now%20create%20new%20accounts%20as%20follows.)

File: src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmGenerator.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * IUV code generator with IUV alghoritm interface
+ */
+public interface IuvAlghoritmGenerator {
+
+    /**
+     * Initialization of <code>IuvAlghoritmGenerator</code> class
+     */
+    public static class Builder {
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmGenerator.java]: can we use lombok in some way?

File: src/main/java/it/gov/pagopa/iuvgenerator/GenerateIUV.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            final ExecutionContext context) {
+
+        Logger logger = context.getLogger();
+        
+        IuvGenerationModel body = request.getBody().orElseGet(IuvGenerationModel::new);
+        
+        logger.log(Level.INFO, 
+        		() -> "GenerateIUV function executed at: " + LocalDateTime.now() + " for organizationfiscalcode " + organizationFiscalCode + " with body: "+body);
+
+        if (ObjectUtils.isEmpty(body.getAuxDigit()) || ObjectUtils.isEmpty(body.getSegregationCode())) {
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/iuvgenerator/GenerateIUV.java]: ok, but next time use `javax.validation.Validator`

example: [validate method](https://github.com/pagopa/pagopa-afm-marketplace-be/blob/main/src/main/java/it/pagopa/afm/marketplacebe/config/EntityValidator.java#L34-L47) checks the @NotNull annotation

**Review by jacopocarlini — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmGenerator.java (hunk 3 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmGenerator.java]: done

File: src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmAuxDigit.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvAlghoritmAuxDigit.java]: removed

File: src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvCodeBusiness.java (hunk 2 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/iuvgenerator/producer/IuvCodeBusiness.java]: used lombok builder

**Review by jacopocarlini — APPROVED** [review]
perfect! 😎

**Pull request merged**